web/pandas/pdeps/0007-copy-on-write.md

```diff
@@ -0,0 +1,580 @@
+
+```python
+>>> df2 = df[df['B'] > 3]  # Copy under NumPy's rules
+>>> df2['B'] = 10  # Update (the copy) df2, df not changed
+>>> del df2  # All references to df2 are lost, goes out of scope
+```
+
+And so `df` is not modified. For this reason, the SettingWithCopyWarning was introduced.
+
+_With this proposal_, any result of an indexing operation behaves as a copy
+(Copy-on-Write), and thus chained assignment will _never_ work. Given that there is then
+no ambiguity, the idea is to drop the warning.
+
+The above example is a case where chained assignment doesn't work with current pandas.
+But there are of course also patterns with chained assignment that currently _do_ work
+and are used. _With this proposal_, any chained assignment will not work, and so those
+cases will stop working (e.g. the case above but switching the order):
+
+```python
+>>> df['B'][df['B'] > 3] = 10
+# or
+>>> df['B'][0:5] = 10
+```
+
+These cases will raise a warning ``ChainedAssignmentError``, because they can never
+accomplished what the user intended. There will be false-positive cases when these
+operations are triggered from Cython, because Cython uses a different reference counting
+mechanism. These cases should be rare, since calling pandas code from Cython does not
+have any performance benefits.
+
+### Filtered dataframe
+
+A typical example where the current SettingWithCopyWarning becomes annoying is when
+filtering a DataFrame (which always already returns a copy):
+
+```python
+>>> df = pd.DataFrame({"A": [1, 2], "B": [3, 4], "C": [5, 6]})
+>>> df_filtered = df[df["A"] > 1]
+>>> df_filtered["new_column"] = 1
+SettingWithCopyWarning:
+A value is trying to be set on a copy of a slice from a DataFrame.
+Try using .loc[row_indexer,col_indexer] = value instead
+```
+
+If you then modify your filtered dataframe (e.g. adding a column), you get the
+unnecessary SettingWithCopyWarning (with confusing message). The only way to get rid of
+the warning is by doing a defensive copy (`df_filtered = df[df["A"] > 1].copy()`, which
+results in copying the data twice in the current implementation, Copy-on-Write would
+not require ``.copy()`` anymore).
+
+_With this proposal_, the filtered dataframe is never a view and the above
+workflow would work as expected without warning (and thus without needing the extra
+copy).
+
+### Modifying a Series (from DataFrame column)
+
+_Currently_, accessing a column of a DataFrame as a Series is one of the few cases that
+is actually guaranteed to always be a view:
+
+```python
+>>> df = pd.DataFrame({"A": [1, 2], "B": [3, 4], "C": [5, 6]})
+>>> s = df["A"]
+>>> s.iloc[0] = 0   # will also modify df (but no longer with this proposal)
+```
+
+_With this proposal_, any indexing operation results in a copy, so also accessing a
+column as a Series (in practice, it will still be a view of course, but behave as a copy
+through Copy-on-Write). In the above example, mutating `s` will no longer modify the
+parent `df`.
```

Comment:
I think it would be worthwhile to show the way to make this work, i.e., what would a user do if their intent was to modify `df.iloc[0,0]`, but using `"A"` to reference the column.

Reply:
Added a small example for this 

(I changed to use `loc` to avoid the complexity of mixed label/positional indexing; while that is an issue in general when having to use loc or iloc in a single step, it's not strictly related and would only complicate the explanation here, I think)